### ISSUE TITLE
Cosine similarity from tod-bert encodings

### ISSUE THREAD

**Issue by ashishjwr** [jasonwu0731/ToD-BERT#27]
I have a response selection problem where I only want to suggest relevant responses. I wanted to use Cosine similarity between context and response as a threshold to filter out irrelevant responses.
Does it make sense to use encoding of tod-bert-jnt or its fine-tuned version on Response Selection for getting cosine similarity score between context and response to determine relevance? If yes, what should be the threshold? 
I used [tod-bert-jnt](https://huggingface.co/TODBERT/TOD-BERT-JNT-V1) model for computing cosine similarity between couple of context, response pairs but results didn't look good. Similarity score was often above 0.85 even for completely irrelevant examples.
I find it intuitive to use fine-tuned model for cosine similarity since it is using dot product as loss function. I haven't trained fine-tuned model (let me know in case its publicly available) and was wondering if it is worth to give it a try?
